### PR TITLE
Feature/2.x/50 demo home page

### PR DESCRIPTION
**Pull request by finnlewis**

### CODE REVIEW

**Comment by finnlewis** [comment]
Hey @markconroy  - this branch is ready for testing with a the localgov_scarfolk theme, it's been a pain but I think it is almost there now. Just running some tests and I'll get it merged once all pass.

**Comment by finnlewis** [comment]
Tests pass, and the demo content should come out looking pretty much like the demo site where we created it: https://demo.localgovdrupal.org/

If anyone would like to test and approve or comment we can get this out there and crack on!

File: localgov_demo.module [hunk 1]
```diff
@@ -20,6 +21,15 @@ function localgov_demo_modules_installed($modules, $is_syncing) {
     in_array('localgov_demo', $modules)
   ) {
 
+    // Set all alert banners to be live by default.
+    $alert_banner_ids = \Drupal::entityQuery('localgov_alert_banner')->execute();
+    $alert_banner_entities = \Drupal::entityTypeManager()->getStorage('localgov_alert_banner')->loadMultiple($alert_banner_ids);
+
+    foreach ($alert_banner_entities as $alert_banner) {
+      $alert_banner->set('moderation_state', 'published');
```
Comment (stephen-cox) [inline localgov_demo.module:30]: This line looks to be causing an error when enabling the module:

```
In ContentEntityBase.php line 587:
                                      
  [InvalidArgumentException]          
  Field moderation_state is unknown.  
                                      

Exception trace:
  at /app/web/core/lib/Drupal/Core/Entity/ContentEntityBase.php:587
 Drupal\Core\Entity\ContentEntityBase->getTranslatedField() at /app/web/core/lib/Drupal/Core/Entity/ContentEntityBase.php:568
 Drupal\Core\Entity\ContentEntityBase->get() at /app/web/core/lib/Drupal/Core/Entity/ContentEntityBase.php:628
 Drupal\Core\Entity\ContentEntityBase->set() at /app/web/modules/contrib/localgov_demo/localgov_demo.module:29
 localgov_demo_modules_installed() at n/a:n/a
 call_user_func_array() at /app/web/core/lib/Drupal/Core/Extension/ModuleHandler.php:403
 Drupal\Core\Extension\ModuleHandler->invokeAll() at /app/web/core/lib/Drupal/Core/Extension/ModuleInstaller.php:367
 Drupal\Core\Extension\ModuleInstaller->install() at /app/web/core/lib/Drupal/Core/ProxyClass/Extension/ModuleInstaller.php:83
 Drupal\Core\ProxyClass\Extension\ModuleInstaller->install() at /app/vendor/drush/drush/src/Drupal/Commands/pm/PmCommands.php:103
 Drush\Drupal\Commands\pm\PmCommands->enable() at n/a:n/a
 call_user_func_array() at /app/vendor/consolidation/annotated-command/src/CommandProcessor.php:257
 Consolidation\AnnotatedCommand\CommandProcessor->runCommandCallback() at /app/vendor/consolidation/annotated-command/src/CommandProcessor.php:212
 Consolidation\AnnotatedCommand\CommandProcessor->validateRunAndAlter() at /app/vendor/consolidation/annotated-command/src/CommandProcessor.php:176
 Consolidation\AnnotatedCommand\CommandProcessor->process() at /app/vendor/consolidation/annotated-command/src/AnnotatedCommand.php:313
 Consolidation\AnnotatedCommand\AnnotatedCommand->execute() at /app/vendor/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /app/vendor/symfony/console/Application.php:1027
 Symfony\Component\Console\Application->doRunCommand() at /app/vendor/symfony/console/Application.php:273
 Symfony\Component\Console\Application->doRun() at /app/vendor/symfony/console/Application.php:149
 Symfony\Component\Console\Application->run() at /app/vendor/drush/drush/src/Runtime/Runtime.php:118
 Drush\Runtime\Runtime->doRun() at /app/vendor/drush/drush/src/Runtime/Runtime.php:48
 Drush\Runtime\Runtime->run() at /app/vendor/drush/drush/drush.php:72
 require() at /app/vendor/drush/drush/drush:4
```

File: localgov_demo.install [hunk 2]
```diff
@@ -19,59 +19,35 @@ function localgov_demo_install() {
 
   // Set demo front page.
   $system_site = \Drupal::configFactory()->getEditable('system.site');
-  $system_site->set('page.front', '/node/91');
+  $system_site->set('page.front', '/node/107');
```
Comment (stephen-cox) [inline localgov_demo.install]: When I do a site install, node 107 is the The Broads (/national-parks-uk/broads). If NIDs are being maintained then this should be node 67.

**Review by stephen-cox — CHANGES_REQUESTED** [review]
Just a couple of issues. The homepage setting being the most critical.

Otherwise looking good.

File: localgov_demo.module (hunk 1 above)
Comment (stephen-cox) [reply localgov_demo.module:30]: This looks to be due to the profile not including Alert Banner 1.2.0, so not a problem with this module. We either need to do a new release of the profile or change the version requirements as suggested by @Adnan-cds here: https://github.com/localgovdrupal/localgov/issues/365

File: localgov_demo.module [hunk 3]
```diff
@@ -100,13 +121,5 @@ function localgov_demo_modules_installed($modules, $is_syncing) {
       }
     }
 
-    // Regenerate all path aliases.
-    // This needs to happen after entity reference fixes.
-    $nids = \Drupal::entityTypeManager()->getStorage('node')->getQuery()->execute();
-    $nodes = \Drupal::entityTypeManager()->getStorage('node')->loadMultiple($nids);
-    foreach ($nodes as $node) {
-      \Drupal::service('pathauto.generator')->updateEntityAlias($node, 'update');
-    }
```
Comment (stephen-cox) [inline localgov_demo.module]: Any reason why this is has been removed?

File: localgov_demo.install (hunk 2 above)
Comment (finnlewis) [reply localgov_demo.install]: Looks like the node ids are not maintained. Every time we add or remove nodes, the node id for the home page changes. Odd that you got something different though. I will test again.

**Review by stephen-cox — COMMENTED** [review]
Couple more issues.

I'm seeing broken entity references. For example 'Registering a birth', node 11, is no longer associated with 'Births, deaths, marriages and citizenship'.

The links in the services menu don't go to where they should do.

File: localgov_demo.install (hunk 2 above)
Comment (j4-m) [reply localgov_demo.install]: @finn using an entity query to load by UUID would be safer...

**Comment by j4-m** [comment]
I came across this ('quick fact') and I'm unsure if it's expected, seems out-of-place:

Screenshot at 12-03-19

**Comment by finnlewis** [comment]
@j4-m I made that quick fact as an example, so I expect it. Good enough for now, safe enough to try? 

How might we expect it to be less out-of-place?

File: localgov_demo.module (hunk 3 above)
Comment (finnlewis) [reply localgov_demo.module]: Intersting... that looks like a mistake. Thanks!

File: localgov_demo.install (hunk 2 above)
Comment (finnlewis) [reply localgov_demo.install]: Turns out we can just use the relative path https://github.com/localgovdrupal/localgov_demo/pull/59/commits/1a8e91d6cbfc32013f0bda2f659d24cbc5de8108 :)

File: localgov_demo.module (hunk 3 above)
Comment (finnlewis) [reply localgov_demo.module]: Thanks Stevo, fixed this with https://github.com/localgovdrupal/localgov_demo/pull/59/commits/a4e34f200a33c53ea482c74623bb84a1d5d69ccb

File: localgov_demo.module (hunk 1 above)
Comment (finnlewis) [reply localgov_demo.module:30]: Thanks @stephen-cox , this should be resolved with https://github.com/localgovdrupal/localgov/releases/tag/2.1.7

Comment (finnlewis) [reply localgov_demo.module:30]: Does that mean we can resolve this conversation @stephen-cox  ( not sure on protocol! )

**Review by stephen-cox — APPROVED** [review]
All looking good now. Nice work!

**Pull request merged**